kratos/containers/model.cpp

```diff
@@ -228,6 +228,57 @@ namespace Kratos
         return nullptr;
     }
 
+    void Model::save(Serializer& rSerializer) const
+    {
+        //we construct auxiliary arrays to avoid having to serialize sets and maps of unique_ptrs
+        std::vector<VariablesList* > aux_var_lists;
+        std::vector<std::string> aux_names;
+        aux_var_lists.reserve(GetListOfVariableLists().size());
+        aux_names.reserve(mRootModelPartMap.size());
+
+        for(auto it = mRootModelPartMap.begin(); it!=mRootModelPartMap.end(); ++it)
+        {
+            aux_names.push_back(it->first);
+        }
+
+        for(auto it = GetListOfVariableLists().begin(); it!=GetListOfVariableLists().end(); ++it)
+            aux_var_lists.push_back(it->get());
+
+        rSerializer.save("ListOfVariablesLists", aux_var_lists);
+        rSerializer.save("ModelPartNames", aux_names);
+
+        for(auto it = mRootModelPartMap.begin(); it!=mRootModelPartMap.end(); ++it)
+        {
+            rSerializer.save(it->first, (it->second).get());
+        }
+    }
+
+    void Model::load(Serializer& rSerializer)
+    {
+        //we construct auxiliary arrays to avoid having to serialize sets and maps of unique_ptrs
+        std::vector<VariablesList* > aux_var_lists;
+        std::vector<std::string> aux_names;
+
+        rSerializer.load("ListOfVariablesLists", aux_var_lists);
+        rSerializer.load("ModelPartNames", aux_names);
+
+        for(IndexType i=0; i<aux_var_lists.size(); ++i) {
+            auto p_aux_list = std::unique_ptr<VariablesList>(aux_var_lists[i]);
+            GetListOfVariableLists().insert(std::move(p_aux_list)); //NOTE: the ordering may be changed since the pointers are changed, however it should not matter
+        }
+
+        for(IndexType i=0; i<aux_names.size(); ++i) {
+            //NOTE: CreateModelPart CANNOT be used here
```

Comment:
why not? bcs it does not have the arguments needed?

Reply:
well, maybe i should have said "shouldn't" .

the point is that if you create by the CreateModelPart function a new variablelist is added. what the load does is to load the list of variable lists and then use that variable list in creating the model. 
if i used the Create it would add an extra variable list which then would remain orphaned. nothing really would happen but it would be a slight waste of memory

Reply:
mmm, i am afraid my explanation is not too clear. Thing is that variablelists are created for any new modelpart but never deleted. So when you load a model, it loads all of the variablelists you stored, which are the ones you effectively use, then if you create modelparts it will create as many dummy variablelists as calls to the create function. These variable lists are actually never used since you are creating a modelpart and then substituting its variable list with the one you loaded